TombEngine/Game/effects/effects.cpp

```diff
@@ -439,7 +439,172 @@ void TriggerCyborgSpark(int x, int y, int z, short xv, short yv, short zv)
+			spark.life = (GetRandomControl() & 7) + 16;
+			spark.sLife = spark.life;
+			spark.roomNumber = roomNumber;
+		}
+		else
+		{
+			spark.sG = (GetRandomControl() & 0xF) + 32;
+			spark.sB = 0;
+			spark.dR = (GetRandomControl() & 0x3F) + 192;
+			spark.dG = (GetRandomControl() & 0x3F) + 128;
+			spark.dB = 32;
+			spark.colFadeSpeed = 8;
+			spark.fadeToBlack = 16;
+			spark.life = (GetRandomControl() & 7) + 24;
+			spark.sLife = spark.life;
+		}
+
+		spark.extras = unsigned char(extraTrig | ((extras_table[extraTrig] + (GetRandomControl() & 7) + 28) << 3));
+		spark.dynamic = (char)dynamic;
+
+		if (dynamic == -2)
+		{
+			for (i = 0; i < 8; i++)
+			{
+				auto* spark = &Particles[i];
+				auto* dynamic = &ParticleDynamics[spark->dynamic];
+
+				if (dynamic->Flags)
+				{
+					dynamic->On = 1;
+					dynamic->Falloff = 4;
+
+					if (uw == 1)
+						dynamic->Flags = 2;
+					else
+						dynamic->Flags = 1;
+
+					spark->dynamic = (char)i;
+					break;
+				}
+
+				dynamic++;
+			}
+
+			if (i == 8)
+				spark.dynamic = -1;
+		}
+
+		spark.x = (GetRandomControl() & 0x1F) + x - 16;
+		spark.y = (GetRandomControl() & 0x1F) + y - 16;
+		spark.z = (GetRandomControl() & 0x1F) + z - 16;
+		spark.xVel = (GetRandomControl() & 0xFFF) - 2048;
+		spark.yVel = (GetRandomControl() & 0xFFF) - 2048;
+		spark.zVel = (GetRandomControl() & 0xFFF) - 2048;
+
+		if (dynamic != -2 || uw == 1)
+		{
+			spark.x = (GetRandomControl() & 0x1F) + x - 16;
+			spark.y = (GetRandomControl() & 0x1F) + y - 16;
+			spark.z = (GetRandomControl() & 0x1F) + z - 16;
+		}
+		else
+		{
+			spark.x = (GetRandomControl() & 0x1FF) + x - 256;
+			spark.y = (GetRandomControl() & 0x1FF) + y - 256;
+			spark.z = (GetRandomControl() & 0x1FF) + z - 256;
+		}
+
+		if (uw == 1)
+			spark.friction = 17;
+		else
+			spark.friction = 51;
+
+		if (GetRandomControl() & 1)
+		{
+			if (uw == 1)
+				spark.flags = SP_SCALE | SP_DEF | SP_ROTATE | SP_EXPDEF | SP_UNDERWEXP;
+			else
+				spark.flags = SP_SCALE | SP_DEF | SP_ROTATE | SP_EXPDEF;
+
+			spark.rotAng = GetRandomControl() & 0xFFF;
+			spark.rotAdd = (GetRandomControl() & 0xFF) + 128;
+		}
+		else if (uw == 1)
+			spark.flags = SP_SCALE | SP_DEF | SP_EXPDEF | SP_UNDERWEXP;
+		else
+			spark.flags = SP_SCALE | SP_DEF | SP_EXPDEF;
+
+		spark.scalar = 3;
+		spark.gravity = 0;
+		spark.size = (GetRandomControl() & 0xF) + 40;
+		spark.sSize = spark.size * scalar;
+		spark.dSize = spark.size * (scalar + 1);
+		spark.size *= scalar;
+		GetRandomControl();
+		spark.maxYvel = 0;
+
+		if (uw != 2)
+		{
+			if (extraTrig)
+				TriggerExplosionSmoke(x, y, z, uw);
+			else
+				TriggerExplosionSmokeEnd(x, y, z, uw);
+		}
+		else 		
+		{
+			r = spark.sR;
+			g = spark.sG;
+			b = spark.sB;
+			spark.sR = b;
+			spark.sG = r;
+			spark.sB = g;
+
+			r = spark.dR;
+			g = spark.dG;
+			b = spark.dB;
+			spark.dR = b;
+			spark.dG = r;
+			spark.dB = g;
+
+			spark.flags |= SP_PLASMAEXP;
+		}
+
+		//z = 2 * gfMirrorZPlane - z; //TODO: Mirror room
```

Comment:
I'd remove commented mirror code, primarily because most likely mirror will be implemented in a completely different way in TEN, instead of throwing hacks around in every possible place.

Reply:
removed